src/ReferenceNotFoundException.java

```diff
@@ -0,0 +1,16 @@
+@SuppressWarnings("serial")
+public class ReferenceNotFoundException extends WordsException {
```

Comment:
I don't think this is quite right.  It's allowed to retrieve a property an object where that property does not exist (per our discussions and the reference manual).  The result is simply the special value "nothing".  The error should occur only if the user dereferences "nothing".